app/features/dynamo/tools.py

```diff
@@ -27,6 +55,439 @@ def read_text_file(file_path):
+
+                doc = Document(page_content=page_content, metadata=metadata)
+                documents.append(doc)
+
+        except Exception as e:
+                logger.error(f"Failed to load file from {self.file_type} sub loader")
+                logger.error(e)
+
+        return documents
+
+class PptxSubLoader:
+    def __init__(self, file_content: BytesIO, file_type: str):
+        self.file_content = file_content
+        self.file_type = file_type
+
+    def load(self) -> List[Document]:
+        documents = []
+        try:
+            presentation = Presentation(self.file_content)
+            for i, slide in enumerate(presentation.slides):
+                slide_text = []
+                for shape in slide.shapes:
+                    if hasattr(shape, "text"):
+                        slide_text.append(shape.text)
+
+                page_content = "\n".join(slide_text)
+                metadata = {"source": self.file_type, "slide_number": i + 1}
+
+                doc = Document(page_content=page_content, metadata=metadata)
+                documents.append(doc)
+
+        except Exception as e:
+            logger.error("Failed to load file from PPTX sub loader")
+            logger.error(e)
+
+        return documents
+
+class TextSubLoader:
+    def __init__(self, file_content: BytesIO, file_type: str):
+        self.file_content = file_content
+        self.file_type = file_type
+
+    def load(self) -> List[Document]:
+        try:
+            text = self.file_content.read().decode('utf-8')
+            doc = Document(page_content=text, metadata={"source": self.file_type})
+            return [doc]
+        except Exception as e:
+            logger.error(f"Failed to load file from {self.file_type} sub loader")
+            logger.error(e)
+            return []
+
+class JsonSubLoader:
+    def __init__(self, file_content: BytesIO, file_type: str):
+        self.file_content = file_content
+        self.file_type = file_type
+
+    def load(self) -> List[Document]:
+        try:
+            data = json.load(self.file_content)
+            content = json.dumps(data, indent=2)
+            doc = Document(page_content=content, metadata={"source": self.file_type})
+            return [doc]
+        except json.JSONDecodeError as e:
+            logger.error(f"Failed to load JSON file from {self.file_type}: {e}")
+            raise ValueError("Invalid JSON content") from e
+
+class MarkdownSubLoader:
+    def __init__(self, file_content: BytesIO, file_type: str):
+        self.file_content = file_content
+        self.file_type = file_type
+
+    def load(self) -> List[Document]:
+        try:
+            content = self.file_content.read().decode('utf-8')
+            doc = Document(page_content=content, metadata={"source": self.file_type})
+            return [doc]
+        except Exception as e:
+            logger.error(f"Failed to load Markdown file from {self.file_type}")
```

Comment:
Ensure the behavior of the try-except blocks are the same. In this case, it is returning an empty list while other classes' load method is raising a ValueError